compass_app/app/lib/ui/search_form/view_models/search_form_viewmodel.dart

```diff
@@ -24,55 +29,55 @@ class SearchFormViewModel extends ChangeNotifier {
   final _log = Logger('SearchFormViewModel');
   final ContinentRepository _continentRepository;
   final ItineraryConfigRepository _itineraryConfigRepository;
-  List<Continent> _continents = [];
-  String? _selectedContinent;
-  DateTimeRange? _dateRange;
-  int _guests = 0;
+
+  // Immutable data class that holds state the View needs to display
+  SearchFormState searchFormState = SearchFormState();
 
   /// True if the form is valid and can be submitted
   bool get valid =>
-      _guests > 0 && _selectedContinent != null && _dateRange != null;
-
-  /// List of continents.
-  /// Loaded in [load] command.
-  List<Continent> get continents => _continents;
-
-  /// Selected continent.
-  /// Null means no continent is selected.
-  String? get selectedContinent => _selectedContinent;
+      searchFormState.guests > 0 &&
+      searchFormState.selectedContinent != null &&
+      searchFormState.dateRange != null;
 
   /// Set selected continent.
   /// Set to null to clear the selection.
-  set selectedContinent(String? continent) {
-    _selectedContinent = continent;
+  /// A [TypedCommand] that set's the selected continent on the [SearchFormState]
+  void updateSelectedContinent(Continent? continent) {
+    if (continent == searchFormState.selectedContinent) {
+      searchFormState = searchFormState.copy(selectedContinent: null);
+    } else {
+      searchFormState = searchFormState.copy(selectedContinent: continent);
+    }
```

Comment:
I replaced all setters with functions. It's not necessary, but look at the new 'view_command' file I added for context.